public/src/Components/Heroes.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import { JsxElement } from 'typescript';
+import {getAPIResource, getAllHeroes, getHeroeByID, getAllComicsOfHero, getSearchedfHeroes} from '../api/api'
+import ComicsOfHero from './ComicsOfHero';
+import { NavLink } from 'react-router-dom';
+
+const queryString = require('query-string');
+
+
+class Heroes extends React.Component<any, any> {
+    constructor(props:any) {
+        super(props) 
+        this.state = {
+            heroes: [],
+            searchInput: "",
+        } 
+        this.handleChange = this.handleChange.bind(this);
+        this.handleSubmit = this.handleSubmit.bind(this);
+    }
+
+    handleChange(e: any){
+        this.setState({searchInput: e.target.value});
+    }
+    
+    handleSubmit(e: any){
+        e.preventDefault();
+    console.log(this.props);
+    let {searchInput} = this.state;
```

Comment:
why let?

Reply:
and no need to use destructuring here

Reply:
If I try use the state directly, I will receive this error. So I am forced to use destructuring. 
![111](https://user-images.githubusercontent.com/88532865/135907966-632551d8-0641-412a-beea-3359117688b2.PNG)